module/spl/spl-proc.c

```diff
@@ -308,16 +310,19 @@ taskq_seq_show_impl(struct seq_file *f, void *p, boolean_t allflag)
 		if (lheads[i]) {
 			j = 0;
 			list_for_each(lh, lheads[i]) {
+				if (j >= MAX_SHOW_TASK) {
+					seq_printf(f, "\n\t(truncated)");
+					break;
+				}
 				/* show the wait waitq list */
 				if (i == LHEAD_WAIT) {
 					wq = list_entry(lh, wait_queue_t, task_list);
 					if (j == 0)
 						seq_printf(f, "\t%s:",
 						    list_names[i]);
-					else if (j == 12) {
+					else if (j % 8 == 0)
 						seq_printf(f, "\n\t     ");
-						j = 0;
-					}
```

Comment:
It look like the correct functioning of this loop depend on `j` being reset to be 0.  You'd need to modify the `if (j == 0)` conditional as well.